xc/device_common.ml

```diff
@@ -266,6 +270,25 @@ let list_frontends ~xs ?for_devids domid =
                                ) devids)
                  ) kinds)
 
+(* Network SR-IOV VF device *)
+let make_nsv_device ?(back_domid=0) front_domid devid =
+  let frontend = { domid = front_domid; kind = NetSriovVf; devid = devid } in
+  let backend = { domid = back_domid; kind = NetSriovVf; devid = devid } in
+  { backend = backend; frontend = frontend }
+
+let list_xenserver_devices ~xs ?for_devids domid =
```

Comment:
We distinguish between `list_frontends` and `list_backends` for a reason: see the comments in the code. This one looks like a variant `list_frontends`. As mentioned above, it may be useful to just extend `list_frontends` to return sriov devices as well.